anthem/lyrics/records.py

```diff
@@ -28,6 +28,18 @@ def add_xmlid(ctx, record, xmlid, noupdate=False):
     })
 
 
+def create(ctx, model, xmlid, values, noupdate=False):
+    """ Create a record matching xmlid with values """
+    if isinstance(model, basestring):
+        model = ctx.env[model]
+
+    record = ctx.env.ref(xmlid, raise_if_not_found=False)
+    if not record:
+        record = model.create(values)
+        add_xmlid(ctx, record, xmlid, noupdate)
+    return record
```

Comment:
Ideally, we want the songs to be idempotent, so `create` is not a good player in this regard. Why not extending `create_or_update` with a `noupdate` argument?

Reply:
Because a noupdate record shouldn't be updated.

Reply:
BTW, idempotence is guaranteed by the `if not record` condition.

Reply:
:+1: with @guewen You can extending `create_or_update` function.
`noupdate`flag don't indicate that we never want to update a record.
It indicates we don't want update the record automatically with a module update.
Here, you call "manually" the `create_or_update` function, so if you don't want to update an existing record, just don't call the `create_or_update` function if record already exists.

Also, I would like to understand why you need this feature here.
- If you need a "module" record with `noupdate`, you create it directly in your module with `noupdate`.
- If you need an external record like `__setup__.***`, you just need to check in your script if the record exists before create it.

Reply:
It makes sense to me that a noupdate field can't be updated, I created a dedicated function for that reason. But that's not a big deal, I'll do whatever you want me to do.